airflow/providers/ftp/operators/ftp.py

```diff
@@ -0,0 +1,146 @@
+
+        self.local_filepath_was_str = False
+        if isinstance(local_filepath, str):
+            self.local_filepath = [local_filepath]
+            self.local_filepath_was_str = True
+        else:
+            self.local_filepath = local_filepath
+
+        if isinstance(remote_filepath, str):
+            self.remote_filepath = [remote_filepath]
+        else:
+            self.remote_filepath = remote_filepath
+
+        if len(self.local_filepath) != len(self.remote_filepath):
+            raise ValueError(
+                f'{len(self.local_filepath)} paths in local_filepath '
+                f'!= {len(self.remote_filepath)} paths in remote_filepath'
+            )
+
+        if not (self.operation.lower() == FTPOperation.GET or self.operation.lower() == FTPOperation.PUT):
+            raise TypeError(
+                f"Unsupported operation value {self.operation}, "
+                f"expected {FTPOperation.GET} or {FTPOperation.PUT}."
+            )
+
+    def execute(self, context: Any) -> str | list[str] | None:
+        file_msg = None
+        try:
+            if self.ftp_conn_id and not self.ftp_hook:
+                self.log.info('ftp_hook not provided or invalid. Trying ftp_conn_id to create FTPHook.')
+                self.ftp_hook = FTPHook(ftp_conn_id=self.ftp_conn_id)
+
+            if self.ftp_hook and not isinstance(self.ftp_hook, FTPHook):
+                self.log.info('ftp_hook not provided or invalid. Trying ftp_conn_id to create FTPHook.')
+                self.ftp_hook = FTPHook(ftp_conn_id=self.ftp_conn_id)
+
+            if not self.ftp_hook:
+                raise AirflowException("Cannot operate without ftp_hook or ftp_conn_id.")
+
+            for local_filepath, remote_filepath in zip(self.local_filepath, self.remote_filepath):
+                if self.operation.lower() == FTPOperation.GET:
+                    local_folder = os.path.dirname(local_filepath)
+                    if self.create_intermediate_dirs:
+                        Path(local_folder).mkdir(parents=True, exist_ok=True)
+                    file_msg = f"from {remote_filepath} to {local_filepath}"
+                    self.log.info("Starting to transfer %s", file_msg)
+                    self.ftp_hook.retrieve_file(remote_filepath, local_filepath)
+                else:
+                    remote_folder = os.path.dirname(remote_filepath)
+                    if self.create_intermediate_dirs:
+                        self.ftp_hook.create_directory(remote_folder)
+                    file_msg = f"from {local_filepath} to {remote_filepath}"
+                    self.log.info("Starting to transfer file %s", file_msg)
+                    self.ftp_hook.store_file(remote_filepath, local_filepath)
+
+        except Exception as e:
+            raise AirflowException(f"Error while transferring {file_msg}, error: {str(e)}")
+
+        return self.local_filepath[0] if self.local_filepath_was_str else self.local_filepath
```

Comment:
For simplicity purposes I would return the list `self.local_filepath` regardless of the input. That way you also get rid of `local_filepath_was_str `